src/processPath.ts

```diff
@@ -0,0 +1,19 @@
+import { statSync } from "fs";
+import logger from "../logger";
+import { renameFilesInDirectory, renameSingleFile } from "./renameFiles";
+import { CLIArguments } from "./types";
+import { filePath } from "./types/renameFiles";
+
+export const processPath = async (path: filePath , args : CLIArguments) => {
+    const stats = statSync(path);
```

Comment:
The use of statSync blocks the event loop. Convert this to asynchronous ( `stat` ) to improve performance, especially when handling many files.